README.rst

```diff
@@ -2,3 +2,117 @@ cdist manifestation
 ===================
 
 A helper for writing cdist manifests in Python.
+
+Motivation
+----------
+
+Writing complex manifests in shell can be painful and laborious. With Python and
+a proper abstraction layer, we can write manifests more easily.
+
+Usage
+-----
+
+First of all make sure your manifest is executable and contains
+``#!/usr/bin/env python`` shebang.
+
+Types
+`````
+
+To use type ``__directory``, import ``directory`` (without leading underscores)
+from ``cdist_manifestation.types`` like this:
+
+.. code-block:: python
+
+    from cdist_manifestation.types import directory
+
+Every type is a function accepting a single positional argument, ``object_id``
+which is optional, and arbitrary amount of keyword arguments, which are the
+type parameters.
+
+So, one could create a directory like this:
+
+.. code-block:: python
+
+    from cdist_manifestation.types import directory
+
+    directory('/my/dir', mode=755, owner='me', parents=True)
+
+This will translate to ``__directory /my/dir --mode 755 --owner me --parents``.
+
+There are two noteworthy things in this example:
```

Comment:
This summary is good, but incomplete in the light of the next paragraph. Maybe it would be better to omit it from here and include it at the end of this section, see my comment there

Reply:
The summary was amended and moved at the bottom of this section.